kubermatic-addons/custom-addon/flux2/01-bootstrap.yaml

```diff
@@ -0,0 +1,45 @@
+{{- $httpCredentialsProvided := and (not (empty .Variables.gitRepositoryHttpUsername)) (not (empty .Variables.gitRepositoryHttpPassword)) -}}
+{{- $authenticationRequired := $httpCredentialsProvided -}}
+---
+{{- if $authenticationRequired }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cluster-sync-credentials
+  namespace: flux-system
+data:
+  {{- if $httpCredentialsProvided }}
+  password: {{ .Variables.gitRepositoryHttpUsername | b64enc | quote }}
+  username: {{ .Variables.gitRepositoryHttpPassword | b64enc | quote }}
+  {{- end }}
+{{- end }}
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: GitRepository
+metadata:
+  name: cluster-sync
+  namespace: flux-system
+spec:
+  interval: {{ default "1m" .Variables.updateInterval | quote }}
+  {{- with .Variables.gitRepositoryBranch }}
+  ref:
+    branch: {{ . | quote }}
```

Comment:
`{{ default "main" .Variables. gitRepositoryBranch | quote }}` ? and drop "with"?